[1mcollected 1 item                                                                                                                      
test_data.py::test_kolmogorov_smirnov 0        0.439
1        0.437
2        0.524
3        0.796
4        0.860
         ...
30022    0.845
30023    0.412
30024    0.493
30025    0.372
30026    0.697
Name: danceability, Length: 30027, dtype: float64
0        0.964
1        0.855
2        0.468
3        0.942
4        0.569
         ...
30022    0.768
30023    0.906
30024    0.747
30025    0.974
30026    0.873
Name: energy, Length: 30027, dtype: float64
0          NaN
1       -7.254
2          NaN
3       -7.432
4          NaN
         ...
30022   -5.448
30023      NaN
30024   -9.102
30025   -7.000
30026   -3.189
Name: loudness, Length: 30027, dtype: float64
[31mFAILED
============================================================== FAILURES ===============================================================
[31m[1m_______________________________________________________ test_kolmogorov_smirnov _______________________________________________________
data = (       Unnamed: 0.1  Unnamed: 0  ...                                title                                       text_...      681         681  ...                       NaN                         Make Me Forget
[12869 rows x 20 columns])
    def test_kolmogorov_smirnov(data):
        sample1, sample2 = data
        numerical_columns = [
            "danceability",
            "energy",
            "loudness",
            "speechiness",
            "acousticness",
            "instrumentalness",
            "liveness",
            "valence",
            "tempo",
            "duration_ms"
        ]
        # Let's decide the Type I error probability (related to the False Positive Rate)
        alpha = 0.05
        # Bonferroni correction for multiple hypothesis testing
        # (see my blog post on this topic to see where this comes from:
        # https://towardsdatascience.com/precision-and-recall-trade-off-and-multiple-hypothesis-testing-family-wise-error-rate-vs-false-71a85057ca2b)
        alpha_prime = 1 - (1 - alpha)**(1 / len(numerical_columns))
        for col in numerical_columns:
            # Use the 2-sample KS test (scipy.stats.ks_2sample) on the column
            # col
            print(sample1[col])
            ts, p_value = scipy.stats.ks_2samp(sample1[col], sample2[col], alternative='two-sided')# YOUR CODE HERE
            # Add an assertion so that the test fails if p_value > alpha_prime
            # YOUR CODE HERE
>           assert p_value > alpha_prime
[31m[1mE           assert nan > 0.005116196891823743
[31m[1mtest_data.py[39m[22m:56: AssertionError
[33m========================================================== warnings summary ===========================================================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.MessageMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: Type google._upb._message.ScalarMapContainer uses PyType_Spec with a metaclass that has custom tp_new. This is deprecated and will no longer be allowed in Python 3.14.
../../../../../../../../../../opt/anaconda3/envs/mlflow-60000e2b97642ccc38c7a9ee503977f0abd54819/lib/python3.12/site-packages/wandb/analytics/sentry.py:90
  /opt/anaconda3/envs/mlflow-60000e2b97642ccc38c7a9ee503977f0abd54819/lib/python3.12/site-packages/wandb/analytics/sentry.py:90: SentryHubDeprecationWarning: `sentry_sdk.Hub` is deprecated and will be removed in a future major release. Please consult our 1.x to 2.x migration guide for details on how to migrate `Hub` usage to the new API: https://docs.sentry.io/platforms/python/migration/1.x-to-2.x
    self.hub = sentry_sdk.Hub(client)
../../../../../../../../../../opt/anaconda3/envs/mlflow-60000e2b97642ccc38c7a9ee503977f0abd54819/lib/python3.12/site-packages/google/protobuf/internal/well_known_types.py:174: 3 warnings
test_data.py: 9 warnings
  /opt/anaconda3/envs/mlflow-60000e2b97642ccc38c7a9ee503977f0abd54819/lib/python3.12/site-packages/google/protobuf/internal/well_known_types.py:174: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
    self.FromDatetime(datetime.datetime.utcnow())
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[36m[1m======================================================= short test summary info =======================================================
[31mFAILED[39m test_data.py::[1mtest_kolmogorov_smirnov[22m - assert nan > 0.005116196891823743
[31m=================================================== [1m1 failed[39m[22m, [33m15 warnings[31m in 5.73s ====================================================